main.tf

```diff
@@ -578,6 +578,46 @@ data "aws_iam_policy_document" "agentless_scan_bucket_policy" {
       values   = ["false"]
     }
   }
+
+  statement {
+    sid    = "ForceSSEAES256OnlyUploads"
+    effect = "Deny"
+    actions = [
+      "s3:PutObject"
+    ]
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+    resources = [
+      "${aws_s3_bucket.agentless_scan_bucket[0].arn}/*"
+    ]
+    condition {
+      test     = "StringNotEquals"
+      variable = "s3:x-amz-server-side-encryption"
+      values   = ["AES256"]
```

Comment:
@theopolis after thinking about this, I think we're good on this.  We can always add in `aws:kms` later if we decide to add that capability to the module.

Reply:
@alannix-lw we could also drop this condition and use the other assure that SSE is 'not null'